src/main/java/com/zalando/ECommerceManagement/productManagement/controller/ProductController.java

```
@@ -40,4 +41,17 @@ public List<Product> getProductsByIds(@RequestParam List<Integer> ids) {
     public Product addProduct(@Valid @RequestBody Product product) {
         return productService.createProduct(product);
     }
+
+    @PutMapping("/{id}")
+    public Product updateProduct(@Valid @PathVariable Integer id, @RequestBody Product updateProduct) {
+        Product existingProduct = productService.getProductById(id);
+        if (existingProduct == null) {
```

can you throw this exception in your service